File: CRUD-WALMART-POD3-306672-2316425-873466/inventory-backend/src/main/java/com/vibecoding/inventory/service/InventoryService.java
package com.vibecoding.inventory.service;

import com.vibecoding.inventory.model.InventoryItem;
import com.vibecoding.inventory.repository.InventoryRepository;
import com.vibecoding.inventory.exception.ResourceNotFoundException;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * Service layer for inventory operations.
 */
@Service
@RequiredArgsConstructor
public class InventoryService {
    private static final Logger logger = LoggerFactory.getLogger(InventoryService.class);
    private final InventoryRepository repository;

    public List<InventoryItem> getAllItems() {
        logger.info("Fetching all inventory items");
        return repository.findAll();
    }

    public InventoryItem getItemById(Long id) {
        logger.info("Fetching inventory item with id: {}", id);
        return repository.findById(id).orElseThrow(() -> new ResourceNotFoundException("Item not found with id: " + id));
    }

    public List<InventoryItem> searchItems(String query) {
        logger.info("Searching inventory items with query: {}", query);
        return repository.findByNameContainingIgnoreCaseOrSkuContainingIgnoreCaseOrCategoryContainingIgnoreCase(query, query, query);
    }

    @Transactional
    public InventoryItem createItem(@Valid InventoryItem item) {
        logger.info("Creating new inventory item: {}", item.getName());
        return repository.save(item);
    }

    @Transactional
    public InventoryItem updateItem(Long id, @Valid InventoryItem updatedItem) {
        logger.info("Updating inventory item with id: {}", id);
        InventoryItem item = getItemById(id);
        item.setName(updatedItem.getName());
        item.setSku(updatedItem.getSku());
        item.setCategory(updatedItem.getCategory());
        item.setPrice(updatedItem.getPrice());
        item.setQuantity(updatedItem.getQuantity());
        return repository.save(item);
    }

    @Transactional
    public void deleteItem(Long id) {
        logger.info("Deleting inventory item with id: {}", id);
        if (!repository.existsById(id)) {
            throw new ResourceNotFoundException("Item not found with id: " + id);
        }
        repository.deleteById(id);
    }

    public List<InventoryItem> getLowStockItems(int threshold) {
        logger.info("Fetching items with quantity below threshold: {}", threshold);
        return repository.findByQuantityLessThan(threshold);
    }

    public List<InventoryItem> getOutOfStockItems() {
        logger.info("Fetching items out of stock");
        return repository.findByQuantity(0);
    }
}

File: Prabhu/CRUDDashboard-VisionCopilot-2241061-2094956-2404947/frontend/src/services/inventoryService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api';

export const inventoryService = {
    getDashboardStats: async () => {
        try {
            const response = await axios.get(`${API_URL}/inventory/dashboard`);
            return response.data;
        } catch (error) {
            console.error('Error fetching dashboard stats:', error);
            throw error;
        }
    },
    
    getAllItems: async () => {
        try {
            const response = await axios.get(`${API_URL}/inventory`);
            return response.data;
        } catch (error) {
            console.error('Error fetching inventory items:', error);
            throw error;
        }
    },
    
    getItemById: async (id) => {
        try {
            const response = await axios.get(`${API_URL}/inventory/${id}`);
            return response.data;
        } catch (error) {
            console.error(`Error fetching item with id ${id}:`, error);
            throw error;
        }
    },
    
    searchItems: async (query) => {
        try {
            const response = await axios.get(`${API_URL}/inventory/search?query=${query}`);
            return response.data;
        } catch (error) {
            console.error('Error searching inventory items:', error);
            throw error;
        }
    },
    
    createItem: async (itemData) => {
        try {
            const response = await axios.post(`${API_URL}/inventory`, itemData);
            return response.data;
        } catch (error) {
            console.error('Error creating inventory item:', error);
            throw error;
        }
    },
    
    updateItem: async (id, itemData) => {
        try {
            const response = await axios.put(`${API_URL}/inventory/${id}`, itemData);
            return response.data;
        } catch (error) {
            console.error(`Error updating item with id ${id}:`, error);
            throw error;
        }
    },
    
    deleteItem: async (id) => {
        try {
            await axios.delete(`${API_URL}/inventory/${id}`);
            return true;
        } catch (error) {
            console.error(`Error deleting item with id ${id}:`, error);
            throw error;
        }
    }
};
